client/query/query.go

```diff
@@ -224,14 +222,15 @@ func ConstructLedgerInfoFromChannel(c *channel.Channel) LedgerChannelInfo {
 	}
 }
 
-func ConstructPaymentInfo(c *channel.Channel, vfo *virtualfund.Objective, paid, remaining *big.Int) (PaymentChannelInfo, error) {
+func ConstructPaymentInfo(c *channel.Channel, paid, remaining *big.Int) (PaymentChannelInfo, error) {
 	status := getStatusFromChannel(c)
-	if vfo != nil && vfo.Status == protocols.Completed {
-		// This means intermediaries may not have a fully signed postfund state even though the channel is "ready"
-		// To determine the the correct status we check the status of the virtual fund objective
-
-		status = Ready
+	// ADR 0009 allows for intermediaries to exit the protocol before receiving all signed post funds
+	// So for intermediaries we return ready once they have signed their post fund state
+	amIntermediary := c.MyIndex != 0 && c.MyIndex != uint(len(c.Participants)-1)
+	if amIntermediary && c.PostFundSignedByMe() {
+		status = Open
 	}
```

Comment:
Just a general comment here: can we think of any way to try and keep this code in sync with what's going on in the protocol/objective?

Could we push the "ready to be executed" logic down into a method on the `Channel`/`VirtualChannel` class, and call into that from both i) the protocol and ii) the query package?

We don't need to block this PR on doing that, but if you like the idea we could make an issue or follow it up with a PR.